File: CS/DXSample/ViewModels/ButtonBarItemViewModel.cs
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Windows.Media;

namespace DXSample.ViewModels {

    [POCOViewModel]
    public class ButtonBarItemViewModel : BarItemViewModel {
        Action action;

        public ButtonBarItemViewModel(Action action) {
            Caption = "";
            this.action = action;
        }

        [Command]
        public void ExecuteAction() {
            if (action != null) {
                action();
            }
        }

        public virtual ImageSource LargeGlyph { get; set; }
        public virtual ImageSource SmallGlyph { get; set; }
    }
}

File: VB/DXSample/ViewModels/ButtonBarItemViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports System
Imports System.Windows.Media

Namespace DXSample.ViewModels

    <POCOViewModel>
    Public Class ButtonBarItemViewModel
        Inherits BarItemViewModel

        Private action As Action

        Public Sub New(ByVal action As Action)
            Caption = ""
            Me.action = action
        End Sub

        <Command>
        Public Sub ExecuteAction()
            If action IsNot Nothing Then
                action()
            End If
        End Sub

        Public Overridable Property LargeGlyph As ImageSource

        Public Overridable Property SmallGlyph As ImageSource
    End Class
End Namespace
